### PR TITLE
Database and Taskqueue Master Roles

### PR DESCRIPTION
**Pull request by whoarethebritons**
we should be able to handle them if they are defined until we officially remove them.
https://ocd.appscale.com:8080/job/Daily%20Build/3974/

### CODE REVIEW

**Comment by whoarethebritons** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/3983/
https://ocd.appscale.com:8080/job/Hawkeye-Advanced-4/839/
https://ocd.appscale.com:8080/job/Tools-test/2458/

File: appscale/tools/node_layout.py [hunk 1]
```diff
@@ -531,6 +536,16 @@ def is_valid_advanced_format(self):
     # and return a list of nodes or raise BadConfigurationException.
     nodes = self.distribute_unassigned_roles(node_hash.values(), role_count)
 
+    for node in nodes:
```
Comment (obino) [inline appscale/tools/node_layout.py]: This seems similar to the check in the simple format: should it be its own function/method/library call?

Comment (whoarethebritons) [reply appscale/tools/node_layout.py]: I could make it a part of distribute_unassigned_roles

**Comment by whoarethebritons** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/3985/
https://ocd.appscale.com:8080/job/Tools-test/2461/
https://ocd.appscale.com:8080/job/Hawkeye-Simple-4/850/
https://ocd.appscale.com:8080/job/Hawkeye-Advanced-4/842/
https://ocd.appscale.com:8080/job/Hawkeye-Advanced-8/854/
https://ocd.appscale.com:8080/job/Euca-Faststart/1631/

**Pull request merged**